ingenialink/ethernet/servo.py

```diff
@@ -378,9 +380,24 @@ def _send_mcb_frame(self, cmd, reg, subnode, data=None):
         """
         frame = MCB.build_mcb_frame(cmd, subnode, reg, data)
         self.__lock.acquire()
-        self.socket.sendall(frame)
-        response = self.socket.recv(1024)
-        self.__lock.release()
+        try:
+            try:
+                self.socket.sendall(frame)
+            except socket.error as e:
+                logger.error(f"Error sending data. Reason: {e}")
+                raise ILIOError from e
+            try:
+                response = self.socket.recv(ETH_BUF_SIZE)
+            except socket.timeout as e:
+                logger.error("Timeout while receiving data.")
+                raise ILTimeoutError from e
+            except socket.error as e:
+                logger.error(f"Error receiving data. Reason: {e}")
+                raise ILIOError from e
```

Comment:
I think this error message should be inside the exception:
raise ILIOError("Error receiving data") from e

Same with the others excepts